TESTS.md

```diff
@@ -64,6 +64,93 @@ import A
 import B
 ```
 
+Typeclasses before identifiers
+
+```haskell given
+import Control.Monad.Trans.Control
+       (embed, MonadTransControl, control, MonadBaseControl)
+```
+
+```haskell expect
+import Control.Monad.Trans.Control
+       (MonadBaseControl, MonadTransControl, control, embed)
+```
+
+Typeclasses before operators
+
+```haskell given
+import Data.Monoid ((<>), Monoid)
+```
+
+```haskell expect
+import Data.Monoid (Monoid, (<>))
+```
+
+Typeclass explicit imports are not sorted
+
+```haskell
+import Control.Monad.Trans.Control
+       (MonadBaseControl(restoreM, liftBaseWith))
+```
+
+Data types before identifiers
+
+```haskell given
+import Data.Maybe (catMaybes, Maybe)
+```
+
+```haskell expect
+import Data.Maybe (Maybe, catMaybes)
+```
+
+Data types before operators
+
+```haskell given
+import SomeModule ((<>), Maybe)
+```
+
+```haskell expect
+import SomeModule (Maybe, (<>))
+```
+
+Data types with all imported constructors before identifiers
+
+```haskell given
+import Control.Monad.Trans.Maybe (runMaybeT, MaybeT(..))
+```
+
+```haskell expect
+import Control.Monad.Trans.Maybe (MaybeT(..), runMaybeT)
+```
+
+Data types with explicit imports are not sorted
+
+```haskell
+import Data.Maybe (Maybe(Nothing, Just))
+```
+
+Operators before identifiers
+
+```haskell given
+import Data.Functor ((<$>), void, (<$))
+```
+
+```haskell expect
+import Data.Functor ((<$), (<$>), void)
+```
+
+Big example - capitals first (i.e. typeclasses/types), then operators, then identifiers
+
+```haskell given
+import qualified MegaModule as M ((>>>), MonadBaseControl, void, MaybeT(..), join, Maybe(Nothing, Just), liftIO, Either, (<<<), Monad)
+```
+
+```haskell expect
+import qualified MegaModule as M
+       (Either, Maybe(Nothing, Just), MaybeT(..), Monad, MonadBaseControl,
+        (<<<), (>>>), join, liftIO, void)
+```
```

Comment:
I think the big example is enough to demonstrate this feature. Describing the rules together before or after the big example source code would be better the catch the idea of the sorting rules.